BInaryTreeRightSideView.java

```diff
@@ -0,0 +1,55 @@
+import java.util.*;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+/**
+* Time Complexity: O(N), it has to visit each node in the tree
+* Space Complexity: O(D), where D is the tree's diameter
```

Comment:
The asymptotic space complexity here would be O(n), where n is the total number of nodes in the tree.
In a typical BFS traversal, the maximum amount of nodes/objects the queue holds is all of the last level of the tree. 
For a balanced tree the last row precisely has n/2 nodes, which also happens to be the worst case.